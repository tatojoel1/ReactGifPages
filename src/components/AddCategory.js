import React, { useState } from "react";
import { Form } from "react-bootstrap";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState(""); // ''

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    //console.log("handeSubmit", inputValue);
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <Form.Group controlId="formBasicEmail">
        <p>{inputValue}</p>
        <Form.Label>Type in the box to seek some gifs for you</Form.Label>
        <input
          type="text"
          className="form-control"
          value={inputValue}
          onChange={handleInputChange}
          placeholder="Example...Tacos"
        />
      </Form.Group>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
