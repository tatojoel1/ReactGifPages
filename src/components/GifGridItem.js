import React, { useState, useEffect, useRef } from "react";
import PropTypes from "prop-types";
import { Card, Button, Col, Tooltip, Overlay } from "react-bootstrap";
import { CopyToClipboard } from "react-copy-to-clipboard";

export const GifGridItem = ({ title, url }) => {
  const [copyText, setcopyText] = useState(false);
  const [show, setShow] = useState(false);
  const target = useRef(null);

  useEffect(() => {
    setTimeout(() => {
      setcopyText(false);
      setShow(false);
    }, 2000);
  }, [copyText]);

  return (
    <Col md={4} className="mb-4">
      <Card className="card animate__animated animate__fadeIn">
        <Card.Img variant="top" src={url} alt={title} />
        <Card.Body>
          <Card.Title className="card-title">
            <p>{title}</p>
            {/* {title.length > 0 ? title : "No title found"} */}
          </Card.Title>
        </Card.Body>
        <Card.Text>
          <h4>Link of gif: </h4>
          <a href={url} target="_blank" rel="noopener noreferrer">
            {url}
          </a>
        </Card.Text>
        {/* <Card.Text>{title.length > 0 ? "" : "Sorry, but the author doesn't put it some title in the gif"}</Card.Text> */}
        <Card.Footer>
          <CopyToClipboard text={url} onCopy={() => setcopyText(true)}>
            <Button
              className="copy-button rounded-pill"
              ref={target}
              onClick={() => setShow(!show)}
            >
              Copy
            </Button>
          </CopyToClipboard>
          <Overlay target={target.current} show={show} placement="right">
            {(props) => (
              <Tooltip id="overlay-example" {...props}>
                Copied!
              </Tooltip>
            )}
          </Overlay>
        </Card.Footer>
      </Card>
    </Col>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
