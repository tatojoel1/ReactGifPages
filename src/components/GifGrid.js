import React from "react";
import PropTypes from 'prop-types'
import { CardGroup, Row } from "react-bootstrap";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category);

  return (
    <>
      <CardGroup>
        <h3 className="animate__animated animate__fadeIn"> {category} </h3>

        {loading && <p className="animate__animated animate__flash">Loading</p>}

        <div className="card-grid">
          <Row md={4}>
            {data.map((img) => (
              <GifGridItem key={img.id} {...img} />
            ))}
          </Row>
        </div>
      </CardGroup>
    </>
  );
};

GifGrid.prototypes = {
  category: PropTypes.string.isRequired
}
