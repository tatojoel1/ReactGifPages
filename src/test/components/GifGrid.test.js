import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Prueba en el componente GifGrid", () => {
  const category = "Goku";

  test("El snapshot correspondiente al componente no corresponden", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar items cuando se cargan imagenes con useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "Goku",
      },
      {
        id: "123",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "Smash",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
