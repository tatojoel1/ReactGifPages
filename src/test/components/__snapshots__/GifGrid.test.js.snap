// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente GifGrid Debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <CardGroup>
    <h3
      className="animate__animated animate__fadeIn"
    >
       
      Goku
       
    </h3>
    <div
      className="card-grid"
    >
      <Row
        md={4}
        noGutters={false}
      >
        <GifGridItem
          id="ABC"
          key="ABC"
          title="Goku"
          url="https://localhost/cualquier/cosa.jpg"
        />
        <GifGridItem
          id="123"
          key="123"
          title="Smash"
          url="https://localhost/cualquier/cosa.jpg"
        />
      </Row>
    </div>
  </CardGroup>
</Fragment>
`;

exports[`Prueba en el componente GifGrid El snapshot correspondiente al componente no corresponden 1`] = `
<Fragment>
  <CardGroup>
    <h3
      className="animate__animated animate__fadeIn"
    >
       
      Goku
       
    </h3>
    <p
      className="animate__animated animate__flash"
    >
      Loading
    </p>
    <div
      className="card-grid"
    >
      <Row
        md={4}
        noGutters={false}
      />
    </div>
  </CardGroup>
</Fragment>
`;
